pkg/clusters/addons/kong/kong.go

```diff
@@ -0,0 +1,266 @@
+package kong
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os/exec"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+)
+
+// -----------------------------------------------------------------------------
+// Kong Addon
+// -----------------------------------------------------------------------------
+
+const (
+	// AddonName is the unique name of the Kong cluster.Addon
+	AddonName clusters.AddonName = "kong"
+)
+
+// Addon is a Kong Proxy addon which can be deployed on a clusters.Cluster.
+type Addon struct {
+	namespace  string
+	name       string
+	deployArgs []string
+	dbmode     DBMode
+}
+
+// New produces a new clusters.Addon for Kong but uses a very opionated set of
+// default configurations (see the defaults() function for more details).
+// If you need to customize your Kong deployment, use the kong.Builder instead.
+func New() *Addon {
+	return NewBuilder().Build()
+}
+
+// ProxyURL provides a routable *url.URL for accessing the Kong proxy.
+func (a *Addon) ProxyURL(ctx context.Context, cluster clusters.Cluster) (*url.URL, error) {
+	waitForObjects, ready, err := a.Ready(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	if !ready {
+		return nil, fmt.Errorf("the addon is not ready on cluster %s, see: %+v", cluster.Name(), waitForObjects)
```

Comment:
```suggestion
		return nil, fmt.Errorf("the addon is not ready on cluster %s: non-empty unresolved objects list: %+v", cluster.Name(), waitForObjects)
```

"see X" was not easy to understand for me when I saw this. I had to read some code to understand what X is.
This comment is intended to make understanding the error easier for the reader like me.